pkg/fetcher/message_fetcher.go

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2025 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package fetcher
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/th2-net/th2-common-go/pkg/grpc"
+	"github.com/th2-net/th2-common-go/pkg/log"
+	common "github.com/th2-net/th2-grpc-common-go"
+	lwdp "github.com/th2-net/th2-grpc-lw-data-provider-go"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	LwdpBase64Format     = "BASE_64"
+	LwdpJsonParsedFormat = "JSON_PARSED"
+	lwdpService          = "DataProviderService"
+)
+
+var (
+	logger = log.ForComponent("lwdp-fetcher")
+)
+
+type LwdpFetcher struct {
+	dpClient lwdp.DataProviderClient
+}
+
+func NewLwdpFetcher(router grpc.Router) (*LwdpFetcher, error) {
```

Comment:
note: you can return a value here because `lwdp.DataProviderClient` is an interface and the underlying impl is returned as a pointer. So, the only thing that will be copied if you return a value instead of a pointer is a pointer to `lwdp.DataProviderClient` impl